### PR TITLE
issue #115: handle empty, null or missing organization info

### PR DESCRIPTION
**Pull request by k-allagbe**
- [x] handle empty, null or missing organization info
- [x] tests pass

### CODE REVIEW

**Comment by Francois-Werbrouck** [comment]
Does it also work with other empty entity?

**Comment by k-allagbe** [comment]
> Does it also work with other empty entity?

Will handle those as separate issues if any. But first, please validate the logic.

File: datastore/db/bytebase/FertiScan/update_inspection_function.sql [hunk 1]
```diff
@@ -25,6 +25,11 @@ DECLARE
     organization_info_id uuid;
     location_id uuid;
 BEGIN
+    -- Skip processing if the input JSON object is empty or null
```
Comment (Francois-Werbrouck) [inline datastore/db/bytebase/FertiScan/update_inspection_function.sql:28]: nice

File: datastore/db/bytebase/FertiScan/update_inspection_function.sql [hunk 2]
```diff
@@ -525,9 +534,13 @@ BEGIN
...
         -- Insert organization and get the organization_id
-        INSERT INTO organization (information_id, main_location_id)
-        VALUES (company_info_id, NULL) -- TODO: main_location_id not yet handled
-        RETURNING id INTO organization_id;
+        IF company_info_id IS NOT NULL THEN
+            INSERT INTO organization (information_id, main_location_id)
+            VALUES (company_info_id, NULL) -- TODO: main_location_id not yet handled
+            RETURNING id INTO organization_id;
+        ELSE
+            organization_id := NULL;
```
Comment (Francois-Werbrouck) [inline datastore/db/bytebase/FertiScan/update_inspection_function.sql:542]: Should it be the manufacturer? If both are null, throw a Warning/Error? How do we plan to handle this edge case?

Comment (k-allagbe) [reply datastore/db/bytebase/FertiScan/update_inspection_function.sql:542]: It should be manufacturer. I changed it. 

The company (where the sample has been taken) doesn't affect the fertilizer object itself. So when it is null, it is simply ignored. Unless you had another idea in designing the schema.

**Pull request merged**